lib/IRGen/IRGenMangler.h

```diff
@@ -245,6 +285,24 @@ class IRGenMangler : public Mangle::ASTMangler {
     appendOperator(Op);
     return finalize();
   }
+
+  CanType getCanTypeInContext(const CanType type, IRGenModule *mod) {
+    if (!type->hasArchetype() || type->hasOpenedExistential()) {
+      return type;
+    }
+    assert(mod && "Expected IRGenModule");
+    auto *genEnv = mod->getGenericEnvironment();
+    assert(genEnv && "Expected a GenericEnvironment");
+    CanType canType = type;
+
+    auto *decl = canType->getAnyGeneric();
+    if (decl) {
```

Comment:
It doesn't make sense to check this first -- just map it out of context unconditionally